core/src/parser/grammar.lalrpop

```diff
@@ -521,21 +521,22 @@ LastFieldPat: LastPattern<FieldPattern> = {
 // The right hand side of an `=` inside a destructuring pattern.
 #[inline]
 Pattern: Pattern = {
-    <l: @L> <alias:(<Ident> "@")?> <record_pat: RecordPattern> <r: @R> => {
+    <l: @L> <alias:(<Ident> "@")?> <data: PatternData> <r: @R> => {
         Pattern {
            alias,
-           data: PatternData::Record(record_pat),
+           data,
            span: mk_span(src_id, l, r),
         }
     },
-    Ident => {
-        let span = <>.pos.unwrap();
+};
 
-        Pattern { data: PatternData::Any(<>), alias: None, span }
-    },
+#[inline]
```

Comment:
for some reason we need the inline pragma to get LALRPOP to accept the grammar as non-ambiguous (it was suggested by LALRPOP error message itself).